packages/hyperion-autologging/src/ALSurface.ts

```diff
@@ -71,7 +71,8 @@ export type InitOptions = Types.Options<
     IReactModule: IReact.IReactModuleExports;
     IJsxRuntimeModule: IReact.IJsxRuntimeModuleExports;
     domFlowletAttributeName?: string;
-    channel: ALChannel,
+    // Only used if ALSurfaceMutationPublisher is enabled as well.
+    channel?: ALChannel,
```

Comment:
Just note in the future, we might have to move the definition of `ALChannel` out of this module, but at the same time, it feels odd to have `ALSurface` depend on `ALSurfaceMutationPublisher`.

Reply:
I'm definitely open to doing this a different way.   We could have the publisher enabled within the surface config itself,  like a sub init option,  since they are closely related.   Without surfaces the publisher doesn't make much sense either.

Reply:
I think the way it is currently is ok and you can merge it. I can think that we might have applications that are not react based and they might directly add DOM attributes for the surfaces. In those case, we still want to have publisher work as it is now, but just that we might have different way of handing the surface itself. I don't yet have a strong conviction which way to go. Just wanted to share with you the alternatives to consider. 
But as of now, let's land this and fix the internal issue we have sooner. We can revisit this easily later since this is just a type.